### ISSUE TITLE
clippy false positive on let_and_return

### ISSUE THREAD

**Issue by pm100** [rust-lang/rust-clippy#5972]
clippy warns on this code and suggests a fix. But the fix does not compile.

```
warning: returning the result of a `let` binding from a block
   --> src\bus.rs:140:10
    |
139 |          let x = this.borrow().timerqueue.queue(f,  n, Instant::now() + when);
    |          --------------------------------------------------------------------- unnecessary `let` binding
140 |          x
    |          ^
    |
    = note: `#[warn(clippy::let_and_return)]` on by default
```

trying to compile its suggested fix

```
error[E0597]: `this` does not live long enough
   --> src\bus.rs:139:10
    |
139 |          this.borrow().timerqueue.queue(f,  n, Instant::now() + when)
    |          ^^^^---------
    |          |
    |          borrowed value does not live long enough
    |          a temporary with access to the borrow is created here ...
140 |      }
    |      -
    |      |
    |      `this` dropped here while still borrowed
    |      ... and the borrow might be used here, when that temporary is dropped and runs the destructor for type `std::cell::Ref<'_, bus::_Bus>`

```
although I must say that I am surprised that the compiler complains about this

versions
clippy 0.0.212 (d3fb005a3 2020-07-31)
rustc 1.45.2 (d3fb005a3 2020-07-31)

**Comment by ebroto** [comment]
Thanks for your issue report!

Could you test if this problem persists with the current stable version? It should have been fixed.

**Comment by pm100** [comment]
seems to have fixed it

**Comment by ebroto** [comment]
Glad to hear that, closing this as solved.
